db/arena_wrapped_db_iter.cc

```diff
@@ -18,6 +18,33 @@
 
 namespace rocksdb {
 
+namespace {
+ReadOptions CreateReadOptions(const ReadOptions& read_opts, char* ptr,
+                              const Slice* iter_lb, const Slice* iter_ub,
+                              const Slice* ts) {
+  ReadOptions ret = read_opts;
+  const auto copy_iter_bounds_and_ts = [&ptr, ts](const Slice* bound) {
+    char* saved_addr = ptr;
+    ptr += sizeof(Slice);
+    char* buf = ptr;
+    memcpy(ptr, bound->data(), bound->size());
+    ptr += bound->size();
+    memcpy(ptr, ts->data(), ts->size());
+    ptr += ts->size();
+    Slice* bound_and_ts =
+        new (saved_addr) Slice(buf, bound->size() + ts->size());
+    return bound_and_ts;
+  };
+  if (iter_lb) {
+    ret.iterate_lower_bound = copy_iter_bounds_and_ts(iter_lb);
+  }
+  if (iter_ub) {
+    ret.iterate_upper_bound = copy_iter_bounds_and_ts(iter_ub);
+  }
+  return ret;
```

Comment:
Can you explain why we need to make a copy to the upper and lower bounds?
I'm not sure it works with a scenario that upper and lower bound are changed in Seek().

Reply:
This is a valid concern which I did not consider. Will revise the implementation.

Reply:
@siying. Updated the PR and removed the copy of iteration bounds.